src/lib.rs

```diff
@@ -255,9 +255,27 @@ impl<T: PartialEq> PartialEq for Cactus<T> {
         if self.len() != other.len() {
             return false;
         }
-        self.vals()
-            .zip(other.vals())
-            .all(|(x, y)| x == y)
+
+        // This is, in a sense, a manually expanded self.vals().zip(other.vals()) -- doing so
+        // allows us to potentially shortcut the checking of every element using Rc::ptr_eq.
+        let mut si = self.node.as_ref();
```

Comment:
Why `si` and `oi`? Is it "self i" and "other i"? Then why "i"?

Reply:
I guess `sn` and `on` mean "self node" and "other node" too?

Reply:
Since this is basically a `for` loop in disguise, I was using `i` as the standard loop variable name. You're correct on all other fronts.

Reply:
OK. I'll leave it up to you to decide if you want more descriptive names.

Reply:
In this case, I think it's OK personally: it's a nice short function.